### DISCUSSION TITLE
Trouble with n-dimensional arrays

### DISCUSSION THREAD

**Discussion by RoboEyes** [pythonnet/pythonnet#2355] [General]
I just started using pythonnet with C# and I am having trouble passing multi-dimensional arrays back and forth between python and C#. I can receive the arrays in C# but I can not convert the PyObject received to a C# object. Can someone point me to some documentation or example code to help me figure this issue out? To be a little more specific, on the python side I am creating and working with 2, 3 and 4 dimensional matrices that I would like to pass to C#. I can pass the data but I am unable to convert these matrices into C# objects. These matrices are float and integer matrices. Thank you.

**Comment by tebjan** [comment]
If the memory is contiguous, you can use `myPyObject.GetBuffer` in C# and copy the memory into a .NET object.

**Reply by RoboEyes** [reply]
Thank you for your response, that worked.
